basic_pipeline/8_Mixer.md

```diff
@@ -0,0 +1,220 @@
+Here comes the mixer - an element responsible for mixing two streams of frames, coming from two different sources. 
+Once again we start with defining the initialization options and the pads of both types:
+```Elixir
+# FILE: lib/elements/Mixer.ex
+
+defmodule Basic.Elements.Mixer do
+ @moduledoc """
+ The element responsible for mixing the frames coming from two sources, based on their timestamps.
+ """
+ use Membrane.Filter
+ alias Basic.Formats.Frame
+
+ def_input_pad(:first_input, demand_unit: :buffers, caps: {Frame, encoding: :utf8})
+
+ def_input_pad(:second_input, demand_unit: :buffers, caps: {Frame, encoding: :utf8})
+
+ def_output_pad(:output, caps: {Frame, encoding: :utf8})
+ ...
+end
+```
+
+Note, that we have defined two input pads: `:first_input` and the `:second_input`. 
+Each of these input pads will have a corresponding incoming track in form of a buffers stream. That is why we might need a structure that would hold the state of such a track so that we would be able to fetch the last buffer which has appeared on the track as well as a mark that no more are we expecting a buffer to show up on the given input pad. Let's do it in the following way, by defining a `Track` inside the mixer module:
```

Comment:
```suggestion
Each of these input pads will have a corresponding incoming track in form of a buffers stream. That is why we might need a structure that would hold the state of such a track so that we would be able to fetch the last buffer which has appeared on the track as well as mark that we are not expecting buffers on a given pad anymore. Let's do it in the following way, by defining a `Track` inside the mixer module:
```